lib/parser/lexer/literal.rb

```diff
@@ -237,7 +237,7 @@ def coerce_encoding(string)
     end
 
     def clear_buffer
-      @buffer = ''
+      @buffer = ''.dup
```

Comment:
I've recently discovered via @koic that Ruby isn't quite consistent with `String.new`. `String.new.encoding` is `ASCII-8BIT`, whereas `String.new('').encoding` (which doesn't avoid the extra allocation) is `UTF-8`.

I'm happy to change the PR to use `String.new('')` - I presume UTF-8 is the preferred encoding? - I've just found that some library maintainers (particularly Rails) prefer `dup`.